packages/message/src/lib/__tests__/message-service.spec.ts

```diff
@@ -0,0 +1,305 @@
+      sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+      contactId: 'did:key:zKAJCbjwdhuhJBJWHBDSIs',
+      timestamp: new Date(),
+    };
+
+    jest
+      .spyOn(messageService['messageRepository'], 'create')
+      .mockRejectedValueOnce(new Error('Failed to create message'));
+
+    const createEvent = waitForEvent(MessageEventChannel.CreateMessage);
+    messageService.createMessage(newMessage);
+
+    const createdMessageResponse = await createEvent;
+
+    expect(createdMessageResponse).toEqual(
+      expect.objectContaining({
+        status: ServiceResponseStatus.Error,
+        payload: expect.objectContaining({
+          name: 'Error',
+          message: 'Failed to create message',
+        }),
+      }),
+    );
+  });
+
+  it('should retrieve all messages for a given contact', async () => {
+    const newMessage1: Message = {
+      id: uuidv4(),
+      text: 'Hello, how are you',
+      sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+      contactId: 'did:key:zKAJCbjwdhuhJBJWHBDSIs',
+      timestamp: new Date(),
+    };
+
+    const newMessage2: Message = {
+      id: uuidv4(),
+      text: 'Hello, how are you now?',
+      sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+      contactId: 'did:key:zKAJCbjwdhuhJBJWHBDSIsqwasqwqw',
+      timestamp: new Date(),
+    };
+
+    const newMessage3: Message = {
+      id: uuidv4(),
+      text: 'Hello, how are you today?',
+      sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+      contactId: 'did:key:zKAJCbjwdhuhJBJWHBDSIs',
+      timestamp: new Date(),
+    };
+
+    // Create messages
+    const createEvent1 = waitForEvent(MessageEventChannel.CreateMessage);
+    messageService.createMessage(newMessage1);
+    await createEvent1;
+
+    const createEvent2 = waitForEvent(MessageEventChannel.CreateMessage);
+    messageService.createMessage(newMessage2);
+    await createEvent2;
+
+    const createEvent3 = waitForEvent(MessageEventChannel.CreateMessage);
+    messageService.createMessage(newMessage3);
+    await createEvent3;
+
+    // Retrieve all messages by contactId
+    const getAllEvent = waitForEvent(MessageEventChannel.GetAllByContactId);
+    messageService.getAllMessagesByContact(newMessage1.contactId);
+    const messages = await getAllEvent;
+
+    // expecting just message 1 and message 3
+    expect(messages).toEqual(
+      expect.objectContaining({
+        status: ServiceResponseStatus.Success,
+        payload: expect.arrayContaining([
+          expect.objectContaining({
+            id: newMessage1.id,
+            text: newMessage1.text,
+            sender: newMessage1.sender,
+            contactId: newMessage1.contactId,
+            timestamp: newMessage1.timestamp,
+          }),
+          expect.objectContaining({
+            id: newMessage3.id,
+            text: newMessage3.text,
+            sender: newMessage3.sender,
+            contactId: newMessage3.contactId,
+            timestamp: newMessage3.timestamp,
+          }),
+        ]),
```

Comment:
Any benifit over this? Note that by using `expect.arrayContaining`, it is not certain that newMessage2 is not returned.
```suggestion
        payload: [ newMessage1, newMessage3 ],
```